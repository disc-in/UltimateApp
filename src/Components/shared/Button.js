import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

import theme from '../../styles/theme.style';

const Button = ({ onPress, icon, text, light, small, style, testID }) => {
  const buttonLight = light ? styles.buttonLight : undefined;
  const smallButton = small ? styles.smallButton : undefined;
  const smallIcon = small ? styles.smallIcon : undefined;
  const textLight = light ? styles.textLight : styles.text;

  return (
    <TouchableOpacity
      style={[styles.button, buttonLight, smallButton, style]}
      onPress={onPress}
      testID={testID || 'button'}
    >
      {icon && (
        <Ionicons
          name={icon}
          style={[styles.icon, textLight, smallIcon, text === undefined ? styles.noTextIcon : null]}
        />
      )}
      {text && <Text style={textLight}>{text}</Text>}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    width: '80%',
    height: 40,
    borderRadius: 5,
    backgroundColor: theme.MAIN_COLOR,
    paddingVertical: 10,
    paddingHorizontal: 20,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
  },
  buttonLight: {
    backgroundColor: theme.COLOR_PRIMARY_LIGHT,
    borderColor: theme.MAIN_COLOR,
    borderWidth: 1,
  },
  smallButton: {
    width: 'auto',
    alignSelf: 'center',
    height: 30,
    paddingVertical: 0,
    paddingHorizontal: 10,
  },
  icon: {
    marginRight: 20,
  },
  smallIcon: {
    marginRight: 5,
  },
  noTextIcon: {
    marginRight: 0,
  },
  text: {
    textAlign: 'center',
    color: theme.COLOR_PRIMARY_LIGHT,
    fontSize: theme.FONT_SIZE_MEDIUM,
  },
  textLight: {
    color: theme.MAIN_COLOR,
    fontSize: theme.FONT_SIZE_MEDIUM,
    fontWeight: 'bold',
  },
});

export default Button;
