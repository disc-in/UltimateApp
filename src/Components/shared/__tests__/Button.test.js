import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import Button from '../Button';

describe('<Button />', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<Button text="OK" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when small', () => {
    const { toJSON } = render(<Button text="OK" small />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with an icon', () => {
    const { toJSON } = render(<Button text="OK" icon="share" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when light', () => {
    const { toJSON } = render(<Button text="OK" light />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls delegates onPress prop', async () => {
    const onPress = jest.fn();

    const { getByTestId } = render(<Button text="Easy" onPress={onPress} />);

    await fireEvent.press(getByTestId('button'));

    expect(onPress).toHaveBeenCalled();
  });
});
