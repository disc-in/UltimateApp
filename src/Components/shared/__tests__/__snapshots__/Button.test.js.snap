// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2a9e91",
      "borderRadius": 5,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
      "width": "80%",
    }
  }
  testID="button"
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    OK
  </Text>
</View>
`;

exports[`<Button /> renders correctly when light 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#2a9e91",
      "borderRadius": 5,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
      "width": "80%",
    }
  }
  testID="button"
>
  <Text
    style={
      Object {
        "color": "#2a9e91",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    OK
  </Text>
</View>
`;

exports[`<Button /> renders correctly when small 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#2a9e91",
      "borderRadius": 5,
      "flexDirection": "row",
      "height": 30,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 0,
      "width": "auto",
    }
  }
  testID="button"
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    OK
  </Text>
</View>
`;

exports[`<Button /> renders correctly with an icon 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2a9e91",
      "borderRadius": 5,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 10,
      "width": "80%",
    }
  }
  testID="button"
>
  <Text />
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    OK
  </Text>
</View>
`;
